surefire-its/src/test/java/org/apache/maven/surefire/its/JUnitPlatformRerunFailingTestsIT.java

```diff
@@ -203,6 +203,19 @@ public void testRerunOneTestMethod()
         verifyFailuresOneRetryOneMethod( outputValidator );
     }
 
+    @Test
+    public void testFailOnTooManyFlakes()
+    {
+        OutputValidator outputValidator = unpack().setJUnitVersion( VERSION ).maven().debugLogging()
+            .addGoal( "-Dsurefire.rerunFailingTestsCount=2" )
+            .addGoal( "-Dsurefire.failOnFlakeCount=1" )
```

Comment:
could we add a test for `rerunFailingTestsCount=0` also - or is that tested implicitly, being the default?

Reply:
Then it would become a simple test and it should perform as if `failOnFlakeCount` was disabled. Am I right?
So the `failOnFlakeCount` works only with specified `rerunFailingTestsCount > 0`?